src/platform.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Headers
+#include "platform.h"
+#include "utils.h"
+#include <cassert>
+#include <utility>
+
+Platform::File::File(std::string name) :
+#ifdef _WIN32
+		filename(Utils::ToWideString(name))
+#else
+		filename(std::move(name))
+#endif
+{
+	// no-op
+}
+
+bool Platform::File::Exists() const {
+#ifdef _WIN32
+	return ::GetFileAttributesW(filename.c_str()) != (DWORD)-1;
+#elif (defined(GEKKO) || defined(_3DS) || defined(__SWITCH__))
+	struct stat sb;
+	return ::stat(filename.c_str(), &sb) == 0;
+#elif defined(PSP2)
+	struct SceIoStat sb;
+	return (::sceIoGetstat(filename.c_str(), &sb) >= 0);
+#else
+	return ::access(filename.c_str(), F_OK) != -1;
+#endif
+}
+
+bool Platform::File::IsFile(bool follow_symlinks) const {
+	return GetType(follow_symlinks) == FileType::File;
+}
+
+bool Platform::File::IsDirectory(bool follow_symlinks) const {
+	return GetType(follow_symlinks) == FileType::Directory;
+}
+
+Platform::FileType Platform::File::GetType(bool follow_symlinks) const {
+#if defined(_WIN32)
+	(void)follow_symlinks;
+	int attribs = ::GetFileAttributesW(filename.c_str());
+
+	if (attribs == INVALID_FILE_ATTRIBUTES) {
+		return FileType::Unknown;
+	} else if ((attribs & FILE_ATTRIBUTE_DIRECTORY) == 0) {
+		return FileType::File;
+	} else if ((attribs & (FILE_ATTRIBUTE_DIRECTORY | FILE_ATTRIBUTE_REPARSE_POINT)) == FILE_ATTRIBUTE_DIRECTORY) {
+		return FileType::Directory;
+	}
+	return FileType::Other;
+#elif defined(PSP2)
+	(void)follow_symlinks;
+	struct SceIoStat sb = {};
+	if (::sceIoGetstat(filename.c_str(), &sb) >= 0) {
+		return SCE_S_ISREG(sb.st_mode) != 0 ? FileType::File :
```

Comment:
This one is ugly. the `_S_IS*` macros are meant to be used as bools. Also to me looks a lot cleaner when not written with ternary operators, but that is just preference.

Reply:
Imo ternary is readable, but that's really a matter of taste.
About the != 0: I removed it. Clang tidy complained that I use integers in a boolean context (maybe false positive but doesn't really matter).